### DISCUSSION TITLE
General Slowness and Go to References/Definitions

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [LazyVim/LazyVim#5202] [Q&A]
Anyone else see a dramatic increase in slowness when going to references through fzf or telescope?

In fzf i get timeouts, in telescope it literally takes upwards of 5-6s for the window to appear listing 4-5 items found.

I think I may have to figure out a way to remain on LazyVim 13.x until this is resolved.

I even clean reinstalled LazyVim and the same issue exists, quite slowww.

**Comment by monoira** [comment]
It's instant for me to go to references in typescript using fzf-lua. Maybe you need to re-do your config?

**Comment by josephcrawfordSRH** [comment]
@monoira I am not sure what I would have to re-do honestly.  I tested typescript this morning and it is instant for me as well, however in one of my PHP projects that is where I noticed it being extremely slow.  I will have to see if this is related to intellephense or something else.

**Comment by josephcrawfordSRH** [comment]
I think this has something to do with the introduction of snacks or another new plugin as the profiler shows this and go to references in my php projects were never this slow.

![Screenshot 2024-12-23 at 2 03 21 PM](https://github.com/user-attachments/assets/6542e290-5f56-4db0-b3d4-d3a25e99a189)

![Screenshot 2024-12-23 at 2 04 05 PM](https://github.com/user-attachments/assets/fa50ec0e-6fc6-4a91-9957-423c110603c1)

While I cannot show the source or filenames this shows that `gr` on the method finds 5 instances and it takes around 8+ seconds to find these results.  Something is definitely amiss.  I wish I could figure out how to go back to v13, seems that may not be possible or may cause issues with plugins that have been written to support v14.

Here is to hoping this is resolved soon it hinders my ability to move fast.

**Reply by folke** [reply]
Those fzf lua timings will be wrong. Read the profiler readme.
Same thing for which-key and other plugins that capture input. Waiting for input will be added to the profiling time of that function.

Before you ask, there is no way to know it's blocking, so I cant account for this.

**Reply by josephcrawfordSRH** [reply]
@folke ok I understand what you are saying, however, why is there such a big difference with finding references between v13 and v14?  In v13 even in our largest PHP project finding references only took about a second.  In v14 it won't work in either Telescope or fzf-lua and it would crash.  I then found that I set fzf to async and then it takes a long time as if searching the files is going very slow or something.  When it comes to TypeScript the finding of references is pretty much instant, but with PHP it is not.

I made no config changes between the versions.  Is there a way to set my local to go back to LazyVim 13 so that I am not effected by this?

**Reply by dpetka2001** [reply]
```lua
return {
  "LazyVim/LazyVim",
  version = "13.*",
}
```
should do it

**Reply by josephcrawfordSRH** [reply]
@dpetka2001 in my install lazyvim is installed through the `lua/config/lazy.lua` file as shown below
When I changed the spec line to include the version as you suggested it did not help, 14.x was still installed
```{ "LazyVim/LazyVim", import = "lazyvim.plugins" },```

```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"

if not (vim.uv or vim.loop).fs_stat(lazypath) then
  -- bootstrap lazy.nvim
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable",
    lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

vim.opt.hidden = false
vim.g.vim_json_conceal = 0
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- import any extras modules here
    { import = "lazyvim.plugins.extras.vscode" },
    -- import/override with your plugins
    { import = "plugins" },
  },
  defaults = {
    -- By default, only LazyVim plugins will be lazy-loaded. Your custom plugins will load during startup.
    -- If you know what you're doing, you can set this to `true` to have all your custom plugins lazy-loaded by default.
    lazy = false,
    -- It's recommended to leave version=false for now, since a lot the plugin that support versioning,
    -- have outdated releases, which may break your Neovim install.
    version = false, -- always use the latest git commit
    -- version = "*", -- try installing the latest stable version for plugins that support semver
  },
  install = { colorscheme = { "tokyonight-storm" } },
  checker = { enabled = true }, -- automatically check for plugin updates
  performance = {
    rtp = {
      -- disable some rtp plugins
      disabled_plugins = {
        "gzip",
        -- "matchit",
        -- "matchparen",
        -- "netrwPlugin",
        "tarPlugin",
        "tohtml",
        "tutor",
        "zipPlugin",
      },
    },
  },
})
```

**Reply by dpetka2001** [reply]
Put the above code snippet in a file in `~/.config/nvim/lua/plugins/core.lua` (the filename can be whatever you want, it doesn't really matter). Then open Lazy UI with `<leader>l` and press `C` and `U` to update and restart Neovim.

**Reply by josephcrawfordSRH** [reply]
@dpetka2001 @folke I was incorrect, the issue seems to be elsewhere.  It seems that some of the methods I was jumping to were quick all around, others are very very slow.  I went back as far as LazyVim 12.x and they are still slow even under telescope and some even timeout.

I am guessing this may be something to do with phpactor more than it is LazyVim itself and or another component when it comes to php and finding the references.

**Reply by dpetka2001** [reply]
You might want to take a look at this [phpactor page](https://phpactor.readthedocs.io/en/master/tips/performance.html) and see if that makes any difference for you.

**Comment by josephcrawfordSRH** [comment]
I still have not been able to find the issue, is anyone else seeing this issue in large php projects?
